app/presenters/person_presenter.rb

```diff
@@ -34,14 +31,12 @@ def full_hash_with_profile
   end
 
   def as_json(_options={})
-    attrs = full_hash_with_avatar
-
-    if own_profile? || person_is_following_current_user
-      attrs.merge!(
-        location: @presentable.location,
-        birthday: @presentable.formatted_birthday,
-        bio:      @presentable.bio
-      )
+    attrs = full_hash
+
+    if attrs[:show_profile_info]
+      attrs.merge!(ProfilePresenter.new(profile).private_hash) { |key, v1, v2| v1 }
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.<br>Unused block argument - `v2`. If it's necessary, use `_` or `_v2` as an argument name to indicate that it won't be used.<br>Space between { and | detected.